common/src/main/java/com/wynntils/screens/widgets/SearchWidget.java

```diff
@@ -0,0 +1,94 @@
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.client.resources.sounds.SimpleSoundInstance;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.sounds.SoundEvents;
+
+public class SearchWidget extends TextInputBoxWidget {
+    private final Component DEFAULT_TEXT = new TranslatableComponent("screens.wynntils.searchWidget.defaultSearchText");
+
+    public SearchWidget(
+            int x,
+            int y,
+            int width,
+            int height,
+            Consumer<String> onUpdateConsumer,
+            WynntilsSettingsScreen settingsScreen) {
+        super(x, y, width, height, new TextComponent("Search..."), onUpdateConsumer, settingsScreen);
+    }
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        this.renderBg(poseStack, McUtils.mc(), mouseX, mouseY);
+
+        String cursorChar = getRenderCursorChar();
+
+        FontRenderer.getInstance()
+                .renderAlignedTextInBox(
+                        poseStack,
+                        Objects.equals(textBoxInput, "") && !isFocused()
+                                ? DEFAULT_TEXT.getString()
+                                : (textBoxInput.substring(0, cursorPosition)
+                                        + cursorChar
+                                        + textBoxInput.substring(cursorPosition)),
+                        this.x + 5,
+                        this.x + this.width - 5,
+                        this.y + 6.5f,
+                        this.width,
+                        CommonColors.WHITE,
+                        FontRenderer.TextAlignment.LEFT_ALIGNED,
+                        FontRenderer.TextShadow.NORMAL);
+    }
+
+    @Override
+    protected void renderBg(PoseStack poseStack, Minecraft minecraft, int mouseX, int mouseY) {
+        RenderUtils.drawScalingTexturedRect(
+                poseStack,
+                Texture.SEARCH_BAR.resource(),
+                this.x,
+                this.y,
+                0,
+                this.width,
+                this.height,
+                Texture.SEARCH_BAR.width(),
+                Texture.SEARCH_BAR.height());
+    }
+
+    @Override
+    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+        if (mouseX >= this.x && mouseX <= this.x + this.width && mouseY >= this.y && mouseY <= this.y + this.height) {
+            McUtils.soundManager().play(SimpleSoundInstance.forUI(SoundEvents.UI_BUTTON_CLICK, 1.0F));
+            settingsScreen.setFocusedTextInput(this);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public void updateNarration(NarrationElementOutput narrationElementOutput) {}
+
+    @Override
+    protected void removeFocus() {
+        this.setTextBoxInput("");
```

Comment:
What does this do? Remove the search term if you e.g. tab away from it? Is that really correct?

Reply:
Remove search if you esc away:

Reply:
Could this be undesirable?

Reply:
You can click away and search will stay. Esc for quick search reset